### ISSUE TITLE
"Password Store has stopped" when the auto fill overlay should pop up.

### ISSUE THREAD

**Issue by brainard52** [zeapo/Android-Password-Store#363]
![screenshot_20180101-091651](https://user-images.githubusercontent.com/1023838/34469691-95a95bcc-eed9-11e7-90fb-a9b5bae154e6.png)
![screenshot_20180101-095719](https://user-images.githubusercontent.com/1023838/34469740-33619280-eeda-11e7-893f-a0f5bfee89b2.png)

I am unable to gather logcat stuff because my device isn't rooted and I don't have a USB-C to USB-A cable yet. However, I'm able to replicate this every single time with the following steps:
1. Install app
2. Clone from server (import ssh key, fill out username+server etc.)
3. Enable auto fill
4. Tap in a compatible password field

Please let me know whether there's anything else you would like me to do.

**Comment by zeapo** [comment]
Thanks for the report. This seems like a new issue in android version >= 8. I will let @wongma7 see if there's a way to fix this.

**Comment by brainard52** [comment]
Sounds good. Let me know if I can help test anything.

**Comment by WarodChinosa** [comment]
363

**Comment by tarciosaraiva** [comment]
Facing the same issue here. Happy to help test.

**Comment by brainard52** [comment]
@tarciosaraiva are you rooted? If so, would you grab logcat?

**Comment by tarciosaraiva** [comment]
@brainard52 Unfortunately not. I've tried to install `aLogCat` from F-Droid but couldn't get anywhere. I'll try debugging the APK tonight and see if I can get some logs.

**Comment by tarciosaraiva** [comment]
@brainard52 this is what logcat gives me

```
--------- beginning of crash
01-02 21:13:15.584  2204  2204 E AndroidRuntime: FATAL EXCEPTION: main
01-02 21:13:15.584  2204  2204 E AndroidRuntime: Process: com.zeapo.pwdstore, PID: 2204
01-02 21:13:15.584  2204  2204 E AndroidRuntime: android.view.WindowManager$BadTokenException: Unable to add window android.view.ViewRootImpl$W@27c0a7a -- permission denied for window type 2003
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.view.ViewRootImpl.setView(ViewRootImpl.java:789)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:356)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.app.Dialog.show(Dialog.java:330)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at com.zeapo.pwdstore.autofill.AutofillService.showSelectPasswordDialog(AutofillService.java:508)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at com.zeapo.pwdstore.autofill.AutofillService.onAccessibilityEvent(AutofillService.java:245)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.accessibilityservice.AccessibilityService$2.onAccessibilityEvent(AccessibilityService.java:1524)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.accessibilityservice.AccessibilityService$IAccessibilityServiceClientWrapper.executeMessage(AccessibilityService.java:1710)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:37)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6494)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
01-02 21:13:15.584  2204  2204 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
01-02 21:13:15.601   897   937 I ActivityManager: Showing crash dialog for package com.zeapo.pwdstore u0
```

**Comment by zeapo** [comment]
I see, following the deprecation of [TYPE_SYSTEM_ALERT](https://developer.android.com/reference/android/view/WindowManager.LayoutParams.html#TYPE_SYSTEM_ALERT), I have changed the behavior in ec1ebd4881253e4de8c9c2032d4adfd7db24fd6d and tested it on android 8.0. Seems like I didn't replicate that change elsewhere in the code, see [line 498](https://github.com/zeapo/Android-Password-Store/blob/ec1ebd4881253e4de8c9c2032d4adfd7db24fd6d/app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.java?utf8=%E2%9C%93#L498)

It's an easy fix, but I don't have the time to work on it for the moment. Maybe this weekend, unless someone can make it. However, that person should not start from the HEAD but from the latest tag as some changes have been merged to master but not well tested.

**Comment by zeapo** [comment]
fixed by #364
